### ISSUE TITLE
`fromArbitrary*` considered harmful

### ISSUE THREAD

**Issue by blamario** [Liqwid-Labs/plutus-extra#50]
One issue that ended up bothering me more than the static context is that I ended up with only a single property test. I put it together using the type tetris, and the easiest function to use was `fromArbitrarySpending`. This requires `Arbitrary` constraints on datum and redeemer. The instances should be declared only once per schema, naturally, so the only sane way is to generate all possible inputs. Once the complete generator is plugged into `scriptProperty`, you have to specify exactly which combinations are going to fail and which to pass. Ergo, this single `scriptProperty` call is all you'll ever need and there is no need or point in writing another property test.

From one point of view this is great, but it doesn't match my experience with writing test *suites* in general. Even when dealing with property tests, there's usually more than one property we want to test.

I'm going to rewrite the test to use `GenForSpending` directly, so this is not the end of the world. I think these functions should carry a warning though.

**Comment by kozross** [comment]
@re-xyr Could you add a note to the Haddocks `fromArbitrarySpending` (and `fromArbitraryMinting`) summarizing @blamario's critiques above? This is not a code change, so no version bump is needed.

**Comment by sergesku** [comment]
Starting with `tasty-plutus 5.0` our approach to testing properties has changed somewhat.
The specified `fromArbitrarySpending` and `fromArbitraryMinting` functions no longer exist.
Should we close the issue or do we have to do something with the previous version?

**Comment by kozross** [comment]
If this is no longer relevant in the current version, we can close the issue.
